src/shared/components/action-buttons/CustomiseButton.tsx

```diff
@@ -16,8 +23,64 @@ const CustomiseTable = lazy(
     )
 );
 
+/*
+| Reason  | User clicked               |
+|---------|----------------------------|
+| outside | outside the panel          |
+| button  | top right close (x) button |
+| cancel  | cancel                     |
+| submit  | save                       |
+*/
+type Reason = 'outside' | 'button' | 'cancel' | 'submit';
```

Comment:
```suggestion
type Reason = Parameters<Exclude<ComponentProps<typeof SlidingPanel>['onClose'], undefined>>[0] | 'cancel';
```

I wish we had a better way to type this (like exporting more types from franklin) but for now I guess it allows us to make sure types don't diverge and we are not expecting a string that might be changed in the future in franklin